src/sentry/auth/providers/fly/views.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.auth.providers.oauth2 import OAuth2Login
+from sentry.auth.view import AuthView, ConfigureView
+
+from .client import FlyClient
+from .constants import AUTHORIZE_URL, ERR_NO_ORG_ACCESS, SCOPE
+
+logger = logging.getLogger("sentry.auth.fly")
+
+
+class FlyOAuth2Login(OAuth2Login):
+    authorize_url = AUTHORIZE_URL
+    scope = SCOPE
+
+    def __init__(self, client_id):
+        super().__init__(client_id=client_id)
+
+
+class FetchUser(AuthView):
+    def __init__(self, org=None, *args, **kwargs):
+        """
+        NOTE: org/args are configured via provider `build_config` method and provided at SSO time
+        """
+        self.org = org
+        super().__init__(*args, **kwargs)
+
+    def handle(self, request: Request, helper) -> Response:
+        with FlyClient(helper.fetch_state("data")["access_token"]) as client:
+            """
+            Utilize the access token to make final request to token introspection endpoint
+            helper.fetch_state -> base pipeline _fetch_state
+
+            Validate whether the authenticated user is authorized to access the configured SSO org
+            """
+            info = client.get_info()
+            if self.org is not None:
+                data = info["data"]
+                user = data["user"]
```

Comment:
Same as above re. maybe change to `user = info["data"]["user"] (again feel free to ignore)

Reply:
👍 this one is more 🤷 
i suppose i would probably just make this a chain of `.get()`'s before making a chain of brackets.
might switch it to that 🤔